smt/available_solvers.cpp

```diff
@@ -126,8 +126,12 @@ SmtSolver create_solver(SolverEnum se,
   return s;
 }
 
-SmtSolver create_solver_for(
-    SolverEnum se, Engine e, bool logging, bool full_model, bool printing)
+SmtSolver create_solver_for(SolverEnum se,
+                            Engine e,
+                            bool logging,
+                            bool full_model,
+                            bool printing,
+                            bool witness)
```

Comment:
I’m considering whether we should pass a `PonoOptions` object here instead, as having so many boolean flags may be error-prone.

Reply:
If we decide to not add `bool witness` (as proposed in https://github.com/stanford-centaur/pono/pull/461#discussion_r2375946204), then we can leave it as is for now.

Reply:
Ended up reverting this.